st2common/tests/unit/services/test_workflow_rerunning.py

```diff
@@ -0,0 +1,230 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+
+import mock
+import os
+import shutil
+import tempfile
+
+from orquesta import statuses as wf_statuses
+
+import st2tests
+
+import st2tests.config as tests_config
+tests_config.parse_args()
+
+from st2common.bootstrap import actionsregistrar
+from st2common.bootstrap import runnersregistrar
+from st2common.constants import action as ac_const
+from st2common.models.db import liveaction as lv_db_models
+from st2common.services import action as ac_svc
+from st2common.services import workflows as wf_svc
+from st2common.transport import liveaction as lv_ac_xport
+from st2common.transport import publishers
+from st2tests.mocks import liveaction as mock_lv_ac_xport
+
+
+TEST_FIXTURES = {
+    'workflows': [
+        'rerun.yaml'
+    ],
+    'actions': [
+        'rerun.yaml'
+    ]
+}
+
+TEST_PACK = 'orquesta_tests'
+TEST_PACK_PATH = st2tests.fixturesloader.get_fixtures_packs_base_path() + '/' + TEST_PACK
+
+PACKS = [
+    TEST_PACK_PATH,
+    st2tests.fixturesloader.get_fixtures_packs_base_path() + '/core'
+]
+
+RETURN_TASK = 'task1'
+OPTIONS = {
+    'tasks': [RETURN_TASK],
+    'reset': [RETURN_TASK]
+}
+
+
+@mock.patch.object(
+    publishers.CUDPublisher,
+    'publish_update',
+    mock.MagicMock(return_value=None))
+@mock.patch.object(
+    publishers.CUDPublisher,
+    'publish_create',
+    mock.MagicMock(side_effect=mock_lv_ac_xport.MockLiveActionPublisher.publish_create))
+@mock.patch.object(
+    lv_ac_xport.LiveActionPublisher,
+    'publish_state',
+    mock.MagicMock(side_effect=mock_lv_ac_xport.MockLiveActionPublisher.publish_state))
+class WorkflowExecutionRerunTest(st2tests.WorkflowTestCase, st2tests.ExecutionDbTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super(WorkflowExecutionRerunTest, cls).setUpClass()
+
+        # Register runners.
+        runnersregistrar.register_runners()
+
+        # Register test pack(s).
+        actions_registrar = actionsregistrar.ActionsRegistrar(
+            use_pack_cache=False,
+            fail_on_failure=True
+        )
+
+        for pack in PACKS:
+            actions_registrar.register_from_pack(pack)
+
+    def setUp(self):
+        super(WorkflowExecutionRerunTest, self).setUp()
+
+        # Create temporary directory used by the tests
+        _, self.temp_dir_path = tempfile.mkstemp()
+        os.chmod(self.temp_dir_path, 0o755)  # nosec
+
+        with open(self.temp_dir_path, 'w') as f:
+            f.write('1\n')
+
+    def tearDown(self):
+        if self.temp_dir_path and os.path.exists(self.temp_dir_path):
+            if os.path.isdir(self.temp_dir_path):
+                shutil.rmtree(self.temp_dir_path)
+            else:
+                os.remove(self.temp_dir_path)
+
```

Comment:
Add a unit tests where 1) the wf_ex_db does not exist and 2) after the rerun, the status is not resuming?